3주차/FindAnimal/src/components/RenderPage.jsx

```diff
@@ -0,0 +1,92 @@
+import { useState, useCallback } from "react";
+import Header from "./Header";
+import Page0 from "../pages/page0";
+import Page1 from "../pages/Page1";
+import Page2 from "../pages/Page2";
+import Page3 from "../pages/Page3";
+import Page4 from "../pages/Page4";
+import PageRandom from "../pages/PageRandom";
+import { ANIMAL_DATA } from "../constants/data";
+import ProgressBar from "./ProgressBar";
+
+const RenderPage = () => {
+  const [showPage, setShowPage] = useState(0);
+  const [selectedOptions, setSelectedOptions] = useState([]);
+
+  const startGame = (page) => {
+    if (page === "page1") {
+      setShowPage(1);
+    } else if (page === "pageRandom") {
+      setShowPage(5);
+    }
+  };
+
+  const restartPage0 = () => {
+    setShowPage(0);
+    setSelectedOptions([]);
+  };
+
+  const restartPage1 = () => {
+    setShowPage(1);
+    setSelectedOptions([]);
+  };
+
+  const handleOptionSelect = useCallback(
+    (option) => {
+      setSelectedOptions((prevOptions) => [...prevOptions, option]);
+    },
+    [setSelectedOptions]
+  );
+
+  const totalPages = 5;
```

Comment:
상수는 전체 대문자 스네이크 케이스로 쓰는게 관행! 
```suggestion
  const TOTAL_PAGES = 5;
```